### DISCUSSION TITLE
custom camera

### DISCUSSION THREAD

**Discussion by hasanshamsijeze** [vsg-dev/VulkanSceneGraph#1323] [Q&A]
Hi I have a problem managing the camera
I need two types of cameras to design a fly simulation 
first: a camera to see above the earth and I can zoom and move the camera to see around.
I don't know how to describe it for example if Camra is in New York, I can have a 360-degree view of around and I can see up or down.
 second: I have a locked camera for air plains that can see from the pilot's point of view.

**Comment by robertosfield** [comment]
What you are looking for is not a custom camera but user defined camera positioning.

You can set the vsg::Camera viewMatrix directly each frame for each vsg::View, or use your own implementation of a camera manipulator, see the vsg::Trackball class used by many of the examples as inspiration.

You can also implement your own subclass from vsg::ViewMatrix to provide the view matrix using your own methods, again have a look at the VSG subclasses for inspiration.
